orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/job/WaitOnJobCompletion.groovy

```diff
@@ -152,21 +160,35 @@ public class WaitOnJobCompletion implements CloudProviderAware, OverridableTimeo
       if ((status == ExecutionStatus.SUCCEEDED) || (status == ExecutionStatus.TERMINAL)) {
         if (stage.context.propertyFile) {
           Map<String, Object> properties = [:]
-          retrySupport.retry({
-            properties = katoRestService.getFileContents(appName, account, location, name, stage.context.propertyFile)
-            if (properties.size() == 0) {
-              if (status == ExecutionStatus.SUCCEEDED) {
-                throw new ConfigurationException("Expected properties file ${stage.context.propertyFile} but it was either missing, empty or contained invalid syntax")
-              } else {
-                throw new ConfigurationException("Expected properties file ${stage.context.propertyFile} but it was either missing, empty or contained invalid syntax (this may be because the job failed)")
-              }
+          try {
+            retrySupport.retry({
+              properties = katoRestService.getFileContents(appName, account, location, name, stage.context.propertyFile)
+            }, 6, 5000, false) // retry for 30 seconds
+          } catch (Exception e) {
+            if (status == ExecutionStatus.SUCCEEDED) {
+              throw new ConfigurationException("Property File: ${stage.context.propertyFile} contents could not be retrieved. Error: " + e)
+            }
+            log.warn("failed to get file contents for ${appName}, account: ${account}, namespace: ${location}, " +
```

Comment:
What would cause this log to be hit?

Reply:
If a jobStatus had `TERMINATED` status. My thinking was at that point, if there is an exception while getting property file contents, then we should not throw that since it will mask the real error regarding why the job was in terminated status. So we just log it and throw the real error a few lines below in the `processFailure(job)` call